docs(sem): add secret deletion (#4851)
scaleway/docs-content: pages/secret-manager/how-to/recover-version.mdx

---
meta:
  title: How to recover a version scheduled for deletion
  description: Recover a version scheduled for deletion in the Scaleway console before it is permanently removed.
content:
  h1: How to recover a version scheduled for deletion
  paragraph: Recover a version scheduled for deletion in the Scaleway console before it is permanently removed.
tags: secret sensitive-data scheduled-deletion recover-version
dates:
  validation: 2025-04-15
  posted: 2025-04-15
categories:
  - identity-and-access-management
---

> This page shows you how to recover a version scheduled for deletion using the Scaleway [console](https://console.scaleway.com). Once you schedule a version for deletion, it enters a 7 day pending deletion period, during which you can still recover it.

Reviewer: ```suggestion
This page shows you how to recover a version scheduled for deletion using the Scaleway [console](https://console.scaleway.com). Once you schedule a version for deletion, it enters a 7-day pending deletion period, during which you can still recover it.
```

After this retention period, the version is permanently deleted.

<Message type="important">
  Scheduled deletion deletes ephemeral secrets and their versions immediately.
</Message>

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/iam/concepts/#owner) status or [IAM permissions](/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- Created a [version](/secret-manager/how-to/create-version/)
- Scheduled a version for deletion

1. Click **Secret Manager** in the **Security and Identity** section of the [Scaleway console](https://console.scaleway.com/) side menu.
2. Select the [region](/secret-manager/concepts/#region) in which to recover the version, in the **Region** drop-down.
3. Access the secret for which you want to recover the version. Your secret's **Overview** tab displays.
4. Click the **Versions** tab.

> 5. Click <Icon name="more" /> next to the version you want to recover, and click **Recover**. A pop-up displays with the estimated cost for recovering the version.

Reviewer: ```suggestion
5. Click <Icon name="more" /> next to the version you want to recover, and click **Recover**. A pop-up displays the estimated cost of recovering the version.
```
